Sources/Kitura/Router.swift

```diff
@@ -18,7 +18,12 @@ import KituraNet
 import KituraSys
 import LoggerAPI
 import Foundation
+
+#if Xcode
+import Kitura_TemplateEngine
+#else
 import KituraTemplateEngine
+#endif
```

Comment:
@aciidb0mb3r What the fix diff was add a more explicit target to the directory structure.

The original repository had source files directly under the Sources directory. In such cases SPM seems to set the module name to the value of the name attribute in the Package.swift file. As the name in the Package.swift file is Kitura-TemplateEngine, the issues you bring up were relevant.

The fix simply added a directory KituraTemplateEngine under the Sources directory and moved the source files under there. That causes SPM to create a module with the name KituraTemplateEngine, whether it is building by itself or generating an XCode project.